### PR TITLE
Improve graph flow rendering algorithm

### PR DESCRIPTION
**Pull request by sigvef**
Before (circled problem areas):
image

After:
image

### CODE REVIEW

File: nin/frontend/app/scripts/directives/editor/GraphEditor.js [hunk 1]
```diff
@@ -35,21 +35,35 @@ class GraphEditor extends React.Component {
...
+          if(source) {
+            depths[source.node.id] = {
+              x: level + 1,
+              y: 0,
+              id: source.node.id,
+              counter
+            };
+            counter++;
+          }
+          if (source) {
```
Comment (aleksanb) [inline nin/frontend/app/scripts/directives/editor/GraphEditor.js]: We can combine these two if's'.

File: nin/frontend/app/scripts/directives/editor/GraphEditor.js [hunk 2]
```diff
@@ -35,21 +35,35 @@ class GraphEditor extends React.Component {
...
             seen.add(source.node.id);
```
Comment (aleksanb) [inline nin/frontend/app/scripts/directives/editor/GraphEditor.js]: A thing i realize now is that we could drop the seen set entirely in favour of using the depths object (the keys would be the same as the contents of the set).

File: nin/frontend/app/scripts/directives/editor/GraphEditor.js [hunk 3]
```diff
@@ -63,6 +77,22 @@ class GraphEditor extends React.Component {
...
+    const positions = [];
+    for(let i = 0; i <= deepestLevel; i++) {
+      positions[i] = [];
+    }
+    for(let nodeId in depths) {
+      positions[depths[nodeId].x].push(depths[nodeId]);
+    }
+
+    for(let x in positions) {
+      console.log(x, positions[x].length);
```
Comment (aleksanb) [inline nin/frontend/app/scripts/directives/editor/GraphEditor.js]: Logging can be removed.

**Comment by sigvef** [comment]
updated

**Pull request merged**